pkg/certificate/manager.go

```diff
@@ -7,21 +7,23 @@ import (
 	"sync"
 	"time"
 
-	"github.com/rs/zerolog/log"
+	"github.com/cskr/pubsub"
 
-	"github.com/openservicemesh/osm/pkg/announcements"
 	"github.com/openservicemesh/osm/pkg/constants"
 	"github.com/openservicemesh/osm/pkg/errcode"
-	"github.com/openservicemesh/osm/pkg/k8s/events"
-	"github.com/openservicemesh/osm/pkg/messaging"
+	"github.com/openservicemesh/osm/pkg/logger"
+)
+
+var (
+	log = logger.New("certificate")
 )
 
 // NewManager creates a new CertificateManager with the passed MRCClient and options
-func NewManager(ctx context.Context, mrcClient MRCClient, getServiceCertValidityPeriod func() time.Duration, getIngressCertValidityDuration func() time.Duration, msgBroker *messaging.Broker, checkInterval time.Duration) (*Manager, error) {
+func NewManager(ctx context.Context, mrcClient MRCClient, getServiceCertValidityPeriod func() time.Duration, getIngressCertValidityDuration func() time.Duration, checkInterval time.Duration) (*Manager, error) {
 	m := &Manager{
 		serviceCertValidityDuration: getServiceCertValidityPeriod,
 		ingressCertValidityDuration: getIngressCertValidityDuration,
-		msgBroker:                   msgBroker,
+		pubsub:                      pubsub.New(1),
```

Comment:
Why not reuse the existing message broker? It uses the same library (github.com/cskr/pubsub)

Reply:
this is about 15 lines of code, and requires 1 of either 2 things:

1. The message broker gets passed around in tandem with the certificate manager everywhere. This is a code smell IMO.
2. The cert manager exposes the message broker itself. 

With the current singleton message broker users still need to DI the broker into the cert manager. We could create another instance, but at that point its an implementation detail, and probably more code to do it that ways anyways.

I'd say why not do it this way, decoupling from another implementation, and [allowing us to move off of pubsub later](https://github.com/openservicemesh/osm/issues/4847)